include/oneapi/tbb/concurrent_queue.h

```diff
@@ -254,13 +261,41 @@ class concurrent_queue {
 
     friend bool operator==( const concurrent_queue<T, Allocator>& lhs,
                             const concurrent_queue<T, Allocator>& rhs ){
-      return lhs.my_queue_representation == rhs.my_queue_representation;
-    }
-  
+        bool ret{false};
+	const_iterator lhs_ptr = lhs.unsafe_begin();
+	const_iterator rhs_ptr = rhs.unsafe_begin();
+	if(lhs.unsafe_size() == rhs.unsafe_size()){
+	    ret = true;
+	    while(lhs_ptr != lhs.unsafe_end()){
+	        if(*lhs_ptr != *rhs_ptr)
+		  ret = false;
+		lhs_ptr++;
+		rhs_ptr++;
+	    }
+	}
+	return ret;
```

Comment:
May be we can simplify using algorithms:

```suggestion
        return lhs.unsafe_size() == rhs.unsafe_size() && std::equal(lhs.unsafe_begin(), lhs.unsafe_end(), rhs.unsafe_begin());
```